### DISCUSSION TITLE
Requesting sitemap.xml on every page once the alternate links added

### DISCUSSION THREAD

**Discussion by AnastasiaTWW** [squidfunk/mkdocs-material#7352] [Q&A]
Hey all! Our docs site built on mkdocs-material is translated to multiple languages. To connect translated pages to each other, we have added alternate links. This is done by adding to the `{% block extrahead %}` block the following code:

```
  {% set current_page = page.url %}
  <link rel="alternate" hreflang="x-default" href="https://docs.wallarm.com/{{ current_page }}">
  <link rel="alternate" hreflang="en" href="https://docs.wallarm.com/{{ current_page }}">
  <link rel="alternate" hreflang="ja" href="https://docs.wallarm.com/ja/{{ current_page }}">
  <link rel="alternate" hreflang="tr" href="https://docs.wallarm.com/tr/{{ current_page }}">
  <link rel="alternate" hreflang="pt-br" href="https://docs.wallarm.com/pt-br/{{ current_page }}">
  <link rel="alternate" hreflang="ar" href="https://docs.wallarm.com/ar/{{ current_page }}">
```

Once we did this, we noticed that requests for sitemap.xml are sent from every page. These requests return a 404 response because the sitemap is only located in the root directory of each language on our site.

I am trying to figure out what the issue is but need your help. You can check it yourself by opening the page https://docs.wallarm.com/user-guides/search-and-filters/custom-report/ and checking the Console output.

How can we fix this? Before adding these alternate links, sitemap.xml was not requested from every page, which is the correct behavior.

**Comment by squidfunk** [comment]
Since this is a customization, we'd definitely need a minimal reproduction to take a look. Without this, we're not able to help, since the error might hide anywhere in your customizations. Also note that we generally cannot help users debug their customizations (we would be doing nothing else all day long), but we can at least check if we can reproduce the error. Thanks!

**Reply by AnastasiaTWW** [reply]
Attaching the minimal reproduction zip. Run the project locally and when you open it in a browser, check the console - it returns 404 for sitemap requests. Thank you for assistance! I will continue investigation on my own as well but probably this is an issue connected with mkdocs-material. On the Wordpress site, we added the same code block (except for variables of course) and it works, requests to sitemap.xml are not sent from every page, so seems like the issue can be connected with tooling

[test-projects-docs.zip](https://github.com/user-attachments/files/16141354/test-projects-docs.zip)
